### PR TITLE
Refactor parts of Date Detector for range/property feature

### PR DESCRIPTION
**Pull request by chiragjn**
This PR is still work in progress, there are other parts which will be merged later from separate branches. Tests will mostly fail, because this PR is incomplete

- disable range and repeat detectors from english
- Add a new meta "dinfo" key to be used later
- Fix some regex and original text bugs
- Refactor Standard date regex for readability and add comments for easier understanding
- Update documentation in Standard date regex
- Add todo and fixme notes for known issues
- Fix timezone exception for time detection (When timezone is passed, the first level converts it to pytz.timezone, resulting in errors in subsequent layers. It uses to fallback to UTC timezone whenever an exception was raised)

### CODE REVIEW

File: ner_v2/detectors/temporal/date/standard_date_regex.py [hunk 1]
```diff
@@ -1,19 +1,22 @@
-# coding=utf-8
+# -*- coding: UTF-8 -*-
+from __future__ import unicode_literals  # All strings in this file are by default unicode literals
 
+import collections
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/standard_date_regex.py]: W1618 import missing `from __future__ import absolute_import`

File: ner_v2/detectors/temporal/time/en/time_detection.py [hunk 2]
```diff
@@ -1,9 +1,8 @@
+import datetime
```
Comment (srajdev) [inline ner_v2/detectors/temporal/time/en/time_detection.py]: W1618 import missing `from __future__ import absolute_import`

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 3]
```diff
@@ -29,10 +27,15 @@ class DateDetector(object):
         tag: entity_name prepended and appended with '__'
         timezone: Optional, pytz.timezone object used for getting current time, default is pytz.timezone('UTC')
         now_date: datetime object holding timestamp while DateDetector instantiation
+<<<<<<< HEAD
+        day_dictionary: dictonary mapping day indexes to day of week spellings and
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: W291 trailing whitespace

File: ner_v2/detectors/temporal/date/standard_date_regex.py [hunk 4]
```diff
@@ -86,458 +73,601 @@ def detect_date(self, text):
...
             date_list.append(date)
             original_list.append(original)
         return date_list, original_list
 
-    def _detect_weekday_ref_month_1(self, date_list, original_list):
+    def _detect_weekday_ref_month_1(self, date_list=None, original_list=None):
         """
-        Parser to detect date containing referenced week and month 'agle month ka pehla monday'(hindi)
+        Detect date using format <number><day of week><next/previous/this/relative delta prepositions>?<month literals>
+            hi -> doosra somwaar agle mahine ka, teesra somwar pichle mahine ka
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/standard_date_regex.py]: W291 trailing whitespace

File: ner_v2/detectors/temporal/date/en/date_detection.py (hunk 3 above)
Comment (chiragjn) [reply ner_v2/detectors/temporal/date/en/date_detection.py]: This is merge error in docstring, will fix it later

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

File: ner_v2/detectors/temporal/date/standard_date_regex.py [hunk 5]
```diff
@@ -1,19 +1,23 @@
...
-from chatbot_ner.config import ner_logger
-from ner_v2.detectors.temporal.constant import TYPE_EXACT
-from ner_v2.detectors.temporal.constant import DATE_CONSTANT_FILE, DATETIME_CONSTANT_FILE, \
-    RELATIVE_DATE, DATE_LITERAL_TYPE, MONTH_LITERAL_TYPE, WEEKDAY_TYPE, \
-    MONTH_TYPE, ADD_DIFF_DATETIME_TYPE, MONTH_DATE_REF_TYPE, NUMERALS_CONSTANT_FILE
-from ner_v2.detectors.temporal.utils import next_weekday, nth_weekday, get_tuple_dict
+import ner_v2.detectors.temporal.constant as temporal_constants
+import ner_v2.detectors.temporal.utils
+import ner_v2.detectors.temporal.utils as temporal_utils
+import ner_v2.detectors.utils as detector_utils
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/standard_date_regex.py]: F401 'ner_v2.detectors.utils as detector_utils' imported but unused

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

File: ner_v2/detectors/temporal/date/standard_date_regex.py [hunk 6]
```diff
@@ -25,519 +27,646 @@ def __init__(self, entity_name, data_directory_path, timezone='UTC', past_date_r
...
-        self.tag = '__' + entity_name + '__'
-        try:
-            self.timezone = pytz.timezone(timezone)
-        except Exception as e:
-            ner_logger.debug('Timezone error: %s ' % e)
-            self.timezone = pytz.timezone('UTC')
-            ner_logger.debug('Default timezone passed as "UTC"')
-
+        self.tag = "__" + entity_name + "__"
+        self.timezone = ner_v2.detectors.temporal.utils.get_timezone(pytz.timezone(timezone))
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/standard_date_regex.py]: F821 undefined name 'ner_v2'

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by chiragjn** [comment]
Closing this, because this is too stale

**Pull request closed without merging**